src/app/api/api_v1/endpoints/test/code_generation_test.py

```diff
@@ -0,0 +1,93 @@
+import unittest
+from typing import Any
+
+import testing.postgresql
+from fastapi.testclient import TestClient
+from sqlalchemy import create_engine, delete
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.ext.asyncio import AsyncSession, create_async_engine
+
+from app.main import app
+from app.api.deps import get_async_db
+from app.db.base import Base
+from app.models.generation_template import GenerationTemplate
+from app.models.welding_configuration import WeldingConfiguration
+from app.models.welding_point import WeldingPoint
+
+
+class GenerationTemplateTest(unittest.TestCase):
+    postgresql: testing.postgresql.Postgresql
+    client: TestClient
+    engine: Any
+    engine_sync: Any
+    template = ("{% for p in welding_points %}"
+                "{{p.x}}, {{p.y}}, {{p.z}} / {{p.roll}}, {{p.pitch}}, {{p.yaw}} / {{p.welding_order}}"
+                "{% endfor %}")
+    session_sync: sessionmaker
+
+    def setUp(self) -> None:
+        self.postgresql = testing.postgresql.Postgresql(port=5678)
+        self.engine_sync = create_engine("postgresql://postgres:awesomepw@localhost:5678", echo=True)
+        self.engine = create_async_engine(
+            "postgresql+asyncpg://postgres:xyz@localhost:5678",
+            echo=True
+        )
+        self.session_sync = sessionmaker(autocommit=False, autoflush=False, bind=self.engine_sync)
+        session_async = sessionmaker(autocommit=False, autoflush=False, bind=self.engine, class_=AsyncSession)
+
+        Base.metadata.create_all(bind=self.engine_sync)
+        self.engine_sync.execute(delete(WeldingPoint))
+        self.engine_sync.execute(delete(WeldingConfiguration))
+        self.engine_sync.execute(delete(GenerationTemplate))
```

Comment:
Do we drop the tables here or just empty them? And why?

Reply:
Tbh I didn't manage to drop them. Very weird behaviour when I try to truncate. The next time I try to access the database the test just [hangs indefinitely](https://github.com/SQE22-RoboticWelding/model-transformer-welding-backend/runs/6812565596?check_suite_focus=true#step:5:1). Debugging/Googling suggests that it might have something to do with the fact that we keep a session in the application open that doesn't allow to truncate. Why then is delete possible if nto truncate? :man_shrugging:

Reply:
Weird, but good to know it works this way. Do you want  to open an issue for that? Just in the case, someone is bored to have another look at it.